### ISSUE TITLE
`normal_` in FlagGems 5.0.x can break same-config reproducibility and appears to drop `torch.Generator`

### ISSUE THREAD

**Issue by tsingmicro-public-e** [FlagOpen/FlagGems#3284]
### Describe the bug

While running a Qwen3 0.6B baseline-aligned short training reproducibility check,
I observed that the same training configuration could not be reproduced across
two reruns after enabling FlagGems. The two runs used the same seed, data/batch
settings, and short exit interval, but the logged metrics diverged at the same
iterations, especially `lm loss`, `grad norm`, and `num zeros`.

In high-level A/B checks, the run was stable with FlagGems disabled, unstable
with FlagGems enabled, and still unstable when FlagCX was disabled / NCCL was used.

The runtime FlagGems log from the unstable full run contained:

```text
flag_gems.ops.normal.normal_: GEMS NORMAL_
```
### Environment

```text
flag_gems==5.0.1rc0
torch==2.7.0+cu126
triton==3.3.0
GEMS_VENDOR=nvidia
device=cuda
```

### Operator-level narrowing

I then ran an operator-level S-only bisect: only a selected set of FlagGems ops
was allowed to use FlagGems, while the other registered FlagGems ops were
disabled. The compared training metrics were `lm loss`, `grad norm`, and
`num zeros`.

The relevant narrowing result was:

```text
normal_ only              unstable
full FlagGems except normal_     stable
full FlagGems except normal_, stop5 confirmation     stable
```

For `normal_ only`, the two reruns diverged immediately:

```text
iter  A_lm_loss      B_lm_loss      A_grad_norm  B_grad_norm
1     1.194329E+01   1.194495E+01   11.779       11.004
2     1.194067E+01   1.195024E+01   17.651       17.909
```
This made `normal_` the main necessary-and-sufficient suspect in this test
configuration.

### Source-level finding

After narrowing the issue to `normal_`, I checked the RNG contract for
`Tensor.normal_(..., generator=gen)`.

The expected behavior is that an explicit `torch.Generator` should control the
random draw and should not consume the default global RNG stream.

However, in the FlagGems `normal_` implementation, the function accepts a
`generator` argument but does not pass it to `normal_distribution`:

```python
def normal_(self, mean=0, std=1, *, generator=None):
    logger.debug("GEMS NORMAL_")
    shape = self.shape
    device = self.device
    self = normal_distribution(shape, device, generator=None, out=self)
    transform_func_float_float(self, std, mean, out0=self)
    return self
```

So the call chain is effectively:

```text
normal_ receives generator
-> normal_distribution is called with generator=None
-> philox_backend_seed_offset(..., generator=None)
-> default global RNG is used instead of the explicit generator
```

I also verified this behavior with a small generator-contract probe. Native
PyTorch passed the checks, while FlagGems failed the explicit-generator cases:

```text
native_same_generator_different_global                  pass  equal=True
native_different_generator_same_global                  pass  equal=False
native_explicit_generator_global_stream_isolated        pass  following_rand_equal=True

gems_same_generator_different_global                    fail  equal=False
gems_different_generator_same_global                    fail  equal=True
gems_explicit_generator_global_stream_isolated          fail  following_rand_equal=False
```

Interpretation:

```text
same explicit generator + different global seed:
  expected equal, FlagGems output differed

different explicit generator + same global seed:
  expected different, FlagGems output was equal

normal_(generator=...) followed by default-stream torch.rand:
  expected default stream unchanged, FlagGems consumed it
```

This looks like a concrete bug in `normal_`: it drops the explicit generator.

### Local patch experiment

I locally changed:

```python
normal_distribution(shape, device, generator=None, out=self)
```

to:

```python
normal_distribution(shape, device, generator=generator, out=self)
```

After that change, the small generator-contract probe passed:

```text
gems_same_generator_different_global                pass    equal=True
gems_different_generator_same_global                pass    equal=False
gems_explicit_generator_global_stream_isolated      pass    following_rand_equal=True
```

However, this did not fully solve the original training-level reproducibility
problem. A full Qwen3 stop1 A/B rerun still diverged:

```text
rerun  iter  lm_loss        grad_norm  num_zeros
A      1     1.194270E+01   17.010     0.0
B      1     1.194494E+01   12.481     1.0
```

So there may be two related issues:

1. A clear source-level bug: `normal_` discards the explicit `generator`.
2. A deeper determinism issue in the 5.0.x `normal_` path, because the training
   repro still diverged even after the local generator-forwarding patch.
